packages/cli/src/cmds/preview/launch.ts

```diff
@@ -99,24 +99,52 @@ export async function openUrlWithNewProfile(url: string): Promise<boolean> {
     if (profileFolderPath === undefined) {
       return false;
     }
-    await open(url, {
-      app: [
-        {
-          name: open.apps.chrome,
-          arguments: [`--user-data-dir=${profileFolderPath}`],
-        },
-        {
-          name: open.apps.edge,
-          arguments: [`--user-data-dir=${profileFolderPath}`],
-        },
-        {
-          name: open.apps.firefox,
-          arguments: ["-profile", profileFolderPath],
-        },
-      ],
-    });
 
-    return true;
+    const tryToOpen = async (
+      url: string,
+      app: { name: string | readonly string[]; arguments: string[] }
+    ) => {
+      return new Promise<boolean>(async (resolve) => {
+        try {
+          const cp = await open(url, {
+            app,
+          });
+          cp.once("close", (code) => {
+            resolve(code === 0);
+          });
+          // NOTE: if app is not existing in the system, open will not throw but cp will exit immediately.
+          // So we may assume that if cp does not exit after 3s, the app is launched successfully.
+          await sleep(3000);
+          if (cp.exitCode !== null && cp.exitCode !== 0) {
+            resolve(false);
+          }
+          resolve(true);
+        } catch {
+          resolve(false);
+        }
+      });
+    };
+
+    const apps = [
+      {
+        name: open.apps.chrome,
+        arguments: [`--user-data-dir=${profileFolderPath}`],
+      },
+      {
+        name: open.apps.edge,
+        arguments: [`--user-data-dir=${profileFolderPath}`],
+      },
+      {
+        name: open.apps.firefox,
+        arguments: ["-profile", profileFolderPath],
+      },
+    ];
+    for (const app of apps) {
```

Comment:
Is that possible to select browser when preview, like the local debug?

Reply:
The browser selected is only for sideloading Teams/Outlook.
For extension part, the browser type cannot be known outside debug provider. So we have to try each browser. And for CLI part, to keep consistent with the extension.